src/rgw/rgw_putobj_processor.cc

```diff
@@ -88,6 +88,8 @@ int RadosWriter::process(bufferlist&& bl, uint64_t offset)
   }
   librados::ObjectWriteOperation op;
   if (offset == 0) {
+    uint64_t len = data.length();
```

Comment:
@cbodley @yehudasa is this commit sufficient, or are there any more places where we know the object size and could hint it?

Reply:
this is sufficient to cover the immutable tail objects

the mutable head object writes happen in `RGWRados::Object::Write::_do_write_meta()` with `op.write_full(*meta.data);` but it's not clear to me whether we should add the hint there too

Reply:
the idea is this would help particularly for small objects - would those only have a head, and no tail objects?

Reply:
exactly right, small object data would always go in the head only. if a head object gets overwritten, we would recreate the object with a new size hint. that's expected to work, right?